subprojects/ide/src/main/java/org/gradle/plugins/ide/eclipse/model/SourceFolder.java

```diff
@@ -58,8 +57,8 @@ private List<String> parseNodeListAttribute(Node node, String attributeName) {
     public SourceFolder(String projectRelativePath, String output) {
         super(projectRelativePath);
         this.output = normalizePath(output);
-        this.includes = Lists.newArrayList();
-        this.excludes = Lists.newArrayList();
+        this.includes = ImmutableList.of();
+        this.excludes = ImmutableList.of();
```

Comment:
This is a potential breaking change. Although it is a bad practice, users might modify the result of `getExcludes()` or `getIncludes()`. We can 1) revert or 2) document the change.

Reply:
I'd revert it in this case.

Reply:
[these](https://github.com/gradle/gradle/blob/ea05cc46c6697b79e2514b85c0804418fae520ef/subprojects/ide/src/main/java/org/gradle/plugins/ide/eclipse/model/SourceFolder.java#L53) are also ending up on that variable, which is also not modifiable.